src/components/ContributionsEpicReminder.tsx

```diff
@@ -125,66 +87,187 @@ const errorTextStyles = css`
     color: ${palette.error.main};
     font-style: italic;
     margin-top: ${space[1]}px;
+    margin-bottom: 0;
 `;
 
-export const ContributionsEpicReminder: React.FC<ReminderFields> = ({
+const getCustomSubmitStyles = (isDisabled: boolean): string | undefined => {
+    if (isDisabled) {
+        // Unfortunately these overrides need !important as otherwise they'll lose
+        // the specificity war against the default Source styles.
+        return css`
+            pointer-events: none;
+            color: ${palette.neutral[60]} !important;
+            background-color: ${palette.neutral[93]} !important;
+            border: 1px solid ${palette.neutral[86]} !important;
+        `;
+    }
+
+    return undefined;
+};
+
+const isValidEmail = (email: string): boolean => {
+    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(String(email).toLowerCase());
+};
+
+interface Props extends ReminderFields {
+    onCloseReminderClick: Function;
+}
+
+type ReminderPayload = {
+    email: string;
+    reminderDate: string;
+};
+
+const contributionsReminderUrl =
+    process.env.NODE_ENV === 'production'
+        ? 'https://contribution-reminders.support.guardianapis.com/remind-me'
+        : 'https://contribution-reminders-code.support.guardianapis.com/remind-me';
+
+const submitForm = ({ email, reminderDate }: ReminderPayload): Promise<any> => {
+    return fetch(contributionsReminderUrl, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+            email,
+            reminderDate,
+            isPreContribution: true,
+        }),
+    });
+};
+
+export const ContributionsEpicReminder: React.FC<Props> = ({
     reminderCTA,
     reminderDate,
     reminderDateAsString,
-}: ReminderFields) => (
-    <div className={rootStyles} data-target="epic-reminder" data-button-copy={reminderCTA}>
-        <div data-target="epic-close" className={closeButtonStyles} tabIndex={0}>
-            <SvgClose />
-        </div>
+    onCloseReminderClick,
+}: Props) => {
+    const [isDirty, setIsDirty] = useState(false);
+    const [isSubmittingState, setIsSubmittingState] = useState(false);
+    const [isErrorState, setIsErrorState] = useState(false);
+    const [isSuccessState, setIsSuccessState] = useState(false);
```

Comment:
As these states should never overlap one another, would it be pertinent to model it more state-machiney?

```js
const defaultFromStates = {
	submitting: false,
	error: false,
	success: false
}

const formStates = {
	default: {...defaultFormStates},
	submitting: {...defaultFormStates, submitting: true},
	error: {...defaultFormStates, error: true},
	success: {...defaultFormStates, success: true}
}

const [formState, setFromState] = useState(formStates.default);

// and then instead of `setIsErrorState
setFormState(formStates.error)

// and use like (not 100% sure on this bit)
const {error, submitting, success} = formState;

{error && 'error state}
```

This way you guard against multiple states set to true when they never should be?

Reply:
Is  there a risk to this approach where if any of the properties on the formStates object change, it will trigger a rerender for all of the others?

Reply:
Yeah, but you probably want that because if error -> success happens you need to rerender elements that rely on `error` and `success` states - whereas before if you `setSuccess(true)` you probably need to be doing `setError(false)`

Reply:
Also, for anything more complex than this, I'd say Formik is well worth a look - https://jaredpalmer.com/formik/

Reply:
I thought about @gtrufitt 's suggestion to handle form states differently and will follow up with a separate PR to test that out.